### PR TITLE
[SYCL] SYCL 2020 standalone device selectors ( gpu_selector_v and friends)

### PR DESCRIPTION
**Pull request by cperkinsintel**
implementation of standalone selectors (e.g. gpu_selector_v, etc)

new tests for llvm-test-suite are here: https://github.com/intel/llvm-test-suite/pull/1141

### CODE REVIEW

**Comment by cperkinsintel** [comment]
/verify with https://github.com/intel/llvm-test-suite/pull/1141

**Comment by cperkinsintel** [comment]
For some reason, when resolving the merge conflicts, there were only conflict in the linux abi .dump file, not the Windows one.  Odd.   I merged, rebuilt on both platforms, and reran the ABI symbols generation on both. The merge conflict is resolved, but on Windows my .dump has picked up the symbols from the recently merged https://github.com/intel/llvm/pull/6474  for some reason. Maybe reordering? Maybe the symbols from 6474, while correct, weren't generated by the python script?

**Comment by cperkinsintel** [comment]
/verify with https://github.com/intel/llvm-test-suite/pull/1141

**Comment by cperkinsintel** [comment]
ping to reviewers.

File: sycl/include/sycl/device_selector.hpp [hunk 1]
```diff
@@ -91,6 +91,14 @@ class __SYCL_EXPORT host_selector : public device_selector {
...
 };
 
+// -------------- SYCL 2020
+
+// SYCL 2020 standalone selectors
+__SYCL_EXPORT int default_selector_v(const device &dev);
+__SYCL_EXPORT int gpu_selector_v(const device &dev);
+__SYCL_EXPORT int cpu_selector_v(const device &dev);
+__SYCL_EXPORT int accelerator_selector_v(const device &dev);
+
```
Comment (aelovikov-intel) [inline sycl/include/sycl/device_selector.hpp:101]: I think we should move it to the top of the file (as that's the most recent/modern interface and we want reader to get familiar with state of the art, not deprecated ways). After that, the section with old-style selectors above should be prefixed with a comment saying that they're remnants of SYCL 1.2.

**Review by aelovikov-intel — APPROVED** [review]
Just one minor nit, the rest LGTM.

**Comment by cperkinsintel** [comment]
ping to reviewers and mergers. I have had to update this twice already because of linux symbol merge conflicts. My worry is that if it takes too long it'll get conflicted again.

**Review by againull — APPROVED** [review]

**Pull request merged**